### PR TITLE
make get_navitia_vjs cacheable

### PR DESCRIPTION
**Pull request by xlqian**

### CODE REVIEW

**Comment by xlqian** [comment]
~2s gain with this PR

**Comment by xlqian** [comment]
Jenkins, test this please!

File: kirin/gtfs_rt/model_maker.py [hunk 1]
```diff
@@ -121,13 +121,13 @@ def _make_trip_updates(self, input_trip_update, data_time):
...
         return trip_updates
 
-    def _get_navitia_vjs(self, trip, data_time):
-        vj_source_code = trip.trip_id
-
-        since = floor_datetime(data_time - self.period_filter_tolerance)
-        until = floor_datetime(data_time + self.period_filter_tolerance + datetime.timedelta(hours=1))
-        self.log.debug('searching for vj {} on [{}, {}] in navitia'.format(vj_source_code, since, until))
+    def __repr__(self):
+	""" Allow this class to be cacheable
```
Comment (xlqian) [inline kirin/gtfs_rt/model_maker.py]: oops

File: kirin/__init__.py [hunk 2]
```diff
@@ -69,6 +69,11 @@
     from gevent import monkey
     monkey.patch_all()
 
+app.config['CACHE_TYPE'] = 'simple'
```
Comment (pbougue) [inline kirin/__init__.py]: Don't you want to default it instead of forcing it?

File: kirin/gtfs_rt/model_maker.py (hunk 1 above)
Comment (pbougue) [inline kirin/gtfs_rt/model_maker.py]: Indent

File: kirin/gtfs_rt/model_maker.py [hunk 3]
```diff
@@ -121,13 +121,13 @@ def _make_trip_updates(self, input_trip_update, data_time):
...
+        """
+        return 'KirinModelBuilder.{}.{}'.format(self.navitia.url, self.contributor)
```
Comment (pbougue) [inline kirin/gtfs_rt/model_maker.py]: Not sure, but you should maybe add `gtfsrt` also (there is another model builder).
Also, maybe use more navitia instance than contributor.

File: kirin/gtfs_rt/model_maker.py [hunk 4]
```diff
@@ -121,13 +121,13 @@ def _make_trip_updates(self, input_trip_update, data_time):
...
-    def _get_navitia_vjs(self, trip, data_time):
-        vj_source_code = trip.trip_id
-
-        since = floor_datetime(data_time - self.period_filter_tolerance)
-        until = floor_datetime(data_time + self.period_filter_tolerance + datetime.timedelta(hours=1))
-        self.log.debug('searching for vj {} on [{}, {}] in navitia'.format(vj_source_code, since, until))
+    def __repr__(self):
+        """ Allow this class to be cacheable
+        """
+        return '{}.{}.{}'.format(self.__class__, self.navitia.url, self.contributor)
```
Comment (xlqian) [inline kirin/gtfs_rt/model_maker.py]: @pbougue you're right. 

1. I use now `self.__class__` to identify the `KirinModelBuilder` class, so the `gtfs` will be included into the cache key
2. I checked `self.navitia.url`, it does contain the coverage/instance name

Comment (pbougue) [inline kirin/gtfs_rt/model_maker.py]: still that use of contrib should probably switch to nav' instance

File: kirin/gtfs_rt/model_maker.py [hunk 5]
```diff
@@ -121,13 +121,13 @@ def _make_trip_updates(self, input_trip_update, data_time):
...
-
-        since = floor_datetime(data_time - self.period_filter_tolerance)
-        until = floor_datetime(data_time + self.period_filter_tolerance + datetime.timedelta(hours=1))
-        self.log.debug('searching for vj {} on [{}, {}] in navitia'.format(vj_source_code, since, until))
+    def __repr__(self):
+        """ Allow this class to be cacheable
+        """
+        return '{}.{}'.format(self.__class__, self.navitia.url)
 
+    @app.cache.memoize(timeout=300)
```
Comment (kinnou02) [inline kirin/gtfs_rt/model_maker.py]: :warning: these object can be upadted, we need to have an invalidation mechanics.

Comment (xlqian) [reply kirin/gtfs_rt/model_maker.py]: @kinnou02 I see

Comment (xlqian) [reply kirin/gtfs_rt/model_maker.py]: that's also why I put a small timeout...

Comment (xlqian) [reply kirin/gtfs_rt/model_maker.py]: @kinnou02 done

File: kirin/gtfs_rt/model_maker.py [hunk 6]
```diff
@@ -121,13 +122,13 @@ def _make_trip_updates(self, input_trip_update, data_time):
...
-    def _get_navitia_vjs(self, trip, data_time):
-        vj_source_code = trip.trip_id
-
-        since = floor_datetime(data_time - self.period_filter_tolerance)
-        until = floor_datetime(data_time + self.period_filter_tolerance + datetime.timedelta(hours=1))
-        self.log.debug('searching for vj {} on [{}, {}] in navitia'.format(vj_source_code, since, until))
+    def __repr__(self):
+        """ Allow this class to be cacheable
+        """
+        return '{}.{}.{}'.format(self.__class__, self.navitia.url, self.instance_data_pub_date)
```
Comment (xlqian) [inline kirin/gtfs_rt/model_maker.py]: @kinnou02

**Comment by buildbotctp** [comment]
Build finished.

File: kirin/gtfs_rt/model_maker.py (hunk 6 above)
Comment (kinnou02) [reply kirin/gtfs_rt/model_maker.py]: It's a vj object from the database, there is no publication date, this object can change at any time

Comment (kinnou02) [reply kirin/gtfs_rt/model_maker.py]: I seems that in fact it isn't a "db_vj" since there is no query to the db in this method, so your cache is ok.

Comment (xlqian) [reply kirin/gtfs_rt/model_maker.py]: yep, you're right, the name of function is not exact, i'll rename it

Comment (xlqian) [reply kirin/gtfs_rt/model_maker.py]: @kinnou02 can you check the PR again plz?

File: tests/mock_navitia/__init__.py [hunk 7]
```diff
@@ -65,3 +65,7 @@ def mock_navitia_query(self, query, q=None):
     mock = _mock_navitia_call[query_str]
 
     return json.loads(mock.json_response), mock.response_code
+
+
+def mock_publication_date(self):
+    return "12345678910"
```
Comment (pbougue) [inline tests/mock_navitia/__init__.py]: Here you are playing a bit with the UNIX 2038 problem... Not sure it's so cool for later tests?

**Pull request merged**